distributed/tests/test_spill.py

```diff
@@ -144,19 +166,22 @@ def test_spillbuffer_maxlim(tmpdir):
         buf["d"] = d
 
     assert "disk reached capacity" in logs_d.getvalue()
-    assert_buf(buf, {"c": c, "d": d, "e": e}, {"a": a, "b": b})
+    assert_buf(buf, tmpdir, {"c": c, "d": d, "e": e}, {"a": a, "b": b})
 
     # Overwrite a key that was in slow, but the size of the new key is larger than
     # max_spill
 
     a_large = "a" * 500
-    assert psize(a_large)[1] > 600  # size of max_spill
+    with TemporaryDirectory() as spill_dir:
+        larger_buf = SpillBuffer(spill_dir, target=0)
+        larger_buf["a_large"] = a_large
+        assert psize(spill_dir, a_large=a_large)[1] > 600  # size of max_spill
```

Comment:
Using TemporaryDirectory in a pytest is anti idiomatic. If you really need a separate temp dir, just have a separate test with the tmpdir fixture.

Reply:
Fair point, I forgot about the `tmpdir_factory` which I can make use of here. I want the additional temp dir within that test to assert the precondition, so I refactored this test to use `tmpdir_factory` instead.